site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,245 @@
+# SQL to Substrait tutorial
+
+To provide an introduction to Substrait for those already familiar with SQL,
+this tutorial walks through the basics of Substrait by translating basic SQL
+queries. We'll present the Substrait in JSON form so it is (somewhat) readable
+for newcomers.
+
+Substrait can be a type, an expression, a relation, or a plan. We'll look at them in
+those order, going from simplest to most complex.
+
+## Types and Schemas
+
+<!-- TODO -->
+
+## Expressions
+
+A basic building block of queries are expressions. Let's start simple with the
+SQL expression:
+
+```sql
+a > 10
+```
+
+An expression can be one of several things. `a` itself will be translated to
+a field reference:
+
+```json
+{
+  "selection": {
+    "directReference": {
+      "structField": {
+        "field": 1
+      }
+    },
+    "rootReference": {}
+  }
+}
+```
+
+!!! note
+    Protobuf may not serialize fields with integer type and value 0, since 0 is the default.
+    So if you instead saw `"structField": {}`, know that is is equivalent to 
+    `"structField": { "field": 0 }`.
+
+`10` will be translated to a literal expression:
+
+```json
+{
+  "literal": {
+    "i8": 10,
```

Comment:
Why is it an `i8`?

Reply:
:shrug: that's what Ibis emitted and it seemed fine to me. Though I suppose usually when people represent an integer it's an i32 or i64, so maybe it would be less weird if it was either of those.

I should probably explain this type to the reader (perhaps that's what you are getting at). One question I have: To what extent should a Substrait producer strive to keep the operands of the same type? If `a` in `a > 10` is an `i64` is it okay if I label `10` as a `i32`?